services/worker/src/worker/job_runners/config/parquet_and_info.py

```diff
@@ -660,8 +650,164 @@ def fill_builder_info(builder: DatasetBuilder, hf_token: Optional[str]) -> None:
                 builder.info.dataset_size += approx_num_bytes
 
 
+class limit_parquet_writes:
```

Comment:
isn't it worth going to the datasets library?

Reply:
(not blocking for this PR)

Reply:
No for several reasons: API doesn't fit the lib, it may creates caching issues, it probably breaks push_to_hub etc.